src/main/java/org/broadinstitute/hellbender/tools/variantdb/ArrayExtractCohortEngine.java

```diff
@@ -0,0 +1,363 @@
+                refAllele.getBases()[0],
+                true,
+                false,
+                true);
+
+
+        final VariantContext finalVC = mergedVC;
+
+        // TODO: this was commented out... probably need to re-enable
+//        final VariantContext annotatedVC = enableVariantAnnotator ?
+//                variantAnnotator.annotateContext(finalizedVC, new FeatureContext(), null, null, a -> true): finalVC;
+
+//        if ( annotatedVC != null ) {
+//            vcfWriter.add(annotatedVC);
+//            progressMeter.update(annotatedVC);
+//        }
+
+        if ( finalVC != null ) {
+            vcfWriter.add(finalVC);
+            progressMeter.update(finalVC);
+        } else {
+            // TODO should i print a warning here?
+            vcfWriter.add(mergedVC);
+            progressMeter.update(mergedVC);
+        }
+    }
+
+    private String formatFloatForVcf(final Float value) {
+        if (value == null || Double.isNaN(value)) {
+            return DOT;
+        }
+        return df.format(value);
+    }
+
+    private Float getNullableFloatFromDouble(Object d) {
+        return d == null ? null : (float)  ((Double) d).doubleValue();
+    }
+
+    private VariantContext createVariantContextFromSampleRecord(final ProbeInfo probeInfo, final GenericRecord sampleRecord, final Set<String> columnNames, final String contig, final long startPosition, final String sample) {
+        final VariantContextBuilder builder = new VariantContextBuilder();
+        final GenotypeBuilder genotypeBuilder = new GenotypeBuilder();
+
+        builder.chr(contig);
+        builder.start(startPosition);
+
+        
+        final List<Allele> alleles = new ArrayList<>();
+        Allele ref = Allele.create(probeInfo.ref, true);        
+        alleles.add(ref);
+
+        Allele alleleA = Allele.create(probeInfo.alleleA, false);
+        Allele alleleB = Allele.create(probeInfo.alleleB, false);
+
+        boolean alleleAisRef = probeInfo.ref.equals(probeInfo.alleleA);
+        boolean alleleBisRef = probeInfo.ref.equals(probeInfo.alleleB);
+
+        if (alleleAisRef) {
+            alleleA = ref;
+        } else {
+            alleles.add(alleleA);
+        }
+
+        if (alleleBisRef) {
+            alleleB = ref;
+        } else {
+            alleles.add(alleleB);
+        }
+
+        builder.alleles(alleles);
+        builder.stop(startPosition + alleles.get(0).length() - 1);
+
+        Float normx;
+        Float normy;
+        Float baf;
+        Float lrr;
+        List<Allele> genotypeAlleles = new ArrayList<Allele>();
+
+        if (this.useCompressedData) {
+            final RawArrayData data = RawArrayData.decode((Long) sampleRecord.get(SchemaUtils.RAW_ARRAY_DATA_FIELD_NAME));
+            normx = data.normx;
+            normy = data.normy;
+            lrr = data.lrr;
+            baf = data.baf;
+
+            // Genotype -- what about no-call?
```

Comment:
is no-call the same as missing? there is a enum value for MISSING.
and if you mean how do you represent a missing allele in the genotype, there is a constant for that. i think it's Genotype.NO_CALL